packages/experience/src/hooks/use-prefilled-identifier.ts

```diff
@@ -0,0 +1,42 @@
+import { type SignInIdentifier } from '@logto/schemas';
+import { useContext, useMemo } from 'react';
+
+import UserInteractionContext from '@/Providers/UserInteractionContextProvider/UserInteractionContext';
+import { type IdentifierInputValue } from '@/components/InputFields/SmartInputField';
+
+import useLoginHint from './use-login-hint';
+
+type Options = {
+  enabledIdentifierType?: SignInIdentifier[];
+};
+
+const usePrefilledIdentifier = ({ enabledIdentifierType }: Options = {}) => {
+  const { identifierInputValue, getIdentifierInputValueByTypes } =
+    useContext(UserInteractionContext);
+
+  const loginHint = useLoginHint();
+
+  return useMemo<IdentifierInputValue>(() => {
+    // First, check if there's a cached input identifier
+    const cachedInputIdentifier = enabledIdentifierType
```

Comment:
When do we don't have the `enabledIdentifierType`? Is it when there's no session storage??

Reply:
Also I think the session storage was introduced for SSO scenario. But it seems even for non-SSO scenarios you are still storing the identifier in the storage. Is it expected?